common/src/main/java/com/wynntils/features/DiscordRichPresenceFeature.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features;
+
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.text.PartStyle;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.ConnectionEvent;
+import com.wynntils.mc.event.SetXpEvent;
+import com.wynntils.models.character.event.CharacterUpdateEvent;
+import com.wynntils.models.character.type.ClassType;
+import com.wynntils.models.territories.profile.TerritoryProfile;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import com.wynntils.utils.mc.McUtils;
+import java.util.Locale;
+import net.minecraft.core.Position;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class DiscordRichPresenceFeature extends Feature {
+    @RegisterConfig
+    public final Config<Boolean> displayLocation = new Config<>(true);
+
+    @RegisterConfig
+    public final Config<Boolean> displayCharacterInfo = new Config<>(true);
+
+    @RegisterConfig
+    public final Config<Boolean> displayWorld = new Config<>(true);
+
+    private boolean stopTerritoryCheck = false;
+    private TerritoryProfile lastTerritoryProfile = null;
+    private ClassType classType = null;
+    private int level = 0;
+
+    private void setCharacterDetails() {
```

Comment:
I don't think it is a formal rule, but I prefer to put the `@SubscribeEvent` methods before private methods, and I think I've noticed @kristofbolyai does the same.